docs/specification.md

```diff
@@ -235,6 +235,133 @@ repository. Clara and Dave are only authorized to work on files within `bar/*`
+succeeds, the new policy is created with the default `allow-rule`.
+
+After a policy is initialized and stored in the gittuf policy namespace, new
+protection rules can be added to the file. In each instance, the policy file is
+re-signed, and therefore, authorized keys for that policy must be presented.
+
+```bash
+$ gittuf policy init
+$ gittuf policy add-rule
+$ gittuf policy remove-rule
+```
+
+Note: the commands listed here are examples and not exhaustive. Please refer to
+gittuf's help documentation for more specific information about gittuf's usage.
+
+### Recording updates in the RSL
+
+The RSL records changes to the policy namespace automatically. To record changes
+to other Git references, the developer must invoke the gittuf client and specify
+the reference. gittuf then examines the reference and creates a new RSL entry.
+
+Similarly, gittuf can also be invoked to create new RSL annotations. In this
+case, the developer must specify the RSL entries the annotation applies to using
+the target entries' Git identifiers.
+
+```bash
+$ gittuf rsl record
+$ gittuf rsl annotate
+```
+
+Note: the commands listed here are examples and not exhaustive. Please refer to
+gittuf's help documentation for more specific information about gittuf's usage.
+
+### Syncing gittuf namespaces with the main repository
+
+gittuf clients uses the `origin` Git remote to identify the main repository. As
+the RSL must be linear with no branches, gittuf employs a variation of the
+`Secure_Fetch` and `Secure_Push` workflows described in the RSL academic paper.
+
+#### RSLFetch: Receiving remote RSL changes
+
+Before local RSL changes can be made or pushed, it is necessary to verify that
+they are compatible with the remote RSL state. If the remote RSL has entries
+that are unavailable locally, entries made locally will be rejected by the
+remote.
+
+TODO: discuss forking attack protections.
+
+1. Fetch remote RSL to the local remote tracker
+   `refs/remotes/origin/gittuf/reference-state-log`.
+1. If the last entry in the remote RSL is the same as the local RSL, terminate
+   successfully.
+1. Perform the verification workflow for the new entries in the remote RSL,
+   incorporating remote changes to the local policy namespace. The verification
+   workflow is performed for each Git reference in the new entries, relative to
+   the local state of each reference. If verification fails, abort and warn
+   user. Note that the verification workflow must fetch each Git reference to
+   its corresponding remote tracker, `refs/remotes/origin/<ref>`. TODO: discuss
+   if verification is skipped for entries that work with namespaces not present
+   locally.
+1. For each modified Git reference, update the local state.
+1. Set local RSL to the remote RSL's tip.
+
+#### RSLPush: Submitting local RSL changes
+
+1. Execute `RSLFetch` repeatedly until there are no new RSL entries in the
+   remote RSL. Every time there is a remote update, the user must be prompted to
```

Comment:
Why is it necessary to prompt the user?  Could this be automated?

Reply:
updated. It could be automated but conflicts make it hard to guarantee always. I suspect this will change a bit once we start implementing, I can note these are drafts until we support them.